sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.net461.cs

```diff
@@ -106,3 +106,23 @@ namespace Azure
         [System.Runtime.CompilerServices.MethodImpl(System.Runtime.CompilerServices.MethodImplOptions.AggressiveInlining)]public bool TryGetValue<T>(out T value) { throw null; }
     }
 }
+namespace Azure.Core
+{
+    public partial class LruCache<TKey, TValue> : System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<TKey, TValue>>, System.Collections.IEnumerable where TKey : notnull
+    {
+        public LruCache(int capacity) { }
+        public int Count { get { throw null; } }
+        public int TotalLength { get { throw null; } }
+        public void AddOrUpdate(TKey key, TValue? val, int length) { }
+        public System.Collections.Generic.IEnumerator<System.Collections.Generic.KeyValuePair<TKey, TValue>> GetEnumerator() { throw null; }
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
+        public bool TryGet(TKey key, out TValue? value) { throw null; }
+    }
+    public abstract partial class SchemaValidator
+    {
+        protected SchemaValidator() { }
+        public abstract string GenerateSchema(System.Type dataType);
+        public abstract bool TryValidate(object data, System.Type dataType, string schemaDefinition, out System.Collections.Generic.IEnumerable<System.Exception> validationErrors);
+        public virtual void Validate(object data, System.Type dataType, string schemaDefinition) { }
```

Comment:
Does this need to be virtual? I wonder if TryValidate would be enough for users.

Reply:
Yeah it doesn't need to be virtual, initially I was thinking we would want both before we had the out param because I wanted to make sure errors could be easily preserved.

Reply:
Is it even helpful to have both?

Reply:
I guess that is the .NET pattern?

Reply:
I'm not sure if there is an explicit pattern for this. The closest that I'm aware of is the pattern we use in DelayStrategy where we have [one abstract method](https://learn.microsoft.com/en-us/dotnet/api/azure.core.delaystrategy.getnextdelaycore?view=azure-dotnet#azure-core-delaystrategy-getnextdelaycore(azure-response-system-int32)), and then a [wrapper non-virtual method](https://learn.microsoft.com/en-us/dotnet/api/azure.core.delaystrategy.getnextdelay?view=azure-dotnet#azure-core-delaystrategy-getnextdelay(azure-response-system-int32)) that calls into the abstract one.

Reply:
In terms of having both Try and non-try, that is definitely a pattern so I think that makes sense to have.

Reply:
I feel like we're coming full circle on the one-or-both.   I think we got here because the architects disliked having only a `Validate` method that threw, and we felt that would be the familiar pattern based on the available libraries.   Personally, I'm fine with both or if we keep just the Try.

Reply:
I like having both - my comment is about whether the Validate method needs to be virtual or if we can leave that up to the library to implement.

Reply:
It would need to be `virtual` for mocking purposes, if nothing else.

Reply:
Yeah I guess for mocking it would need to be virtual, I don't think I expect many customers to be providing their own implementation of it given that they have to implement TryValidate

Reply:
I think I'm going to leave this as-is unless anyone feels strongly about making this non-virtual

Reply:
I don't have a strong opinion, but I suspect leaving this as virtual may cause some confusion for implementers. We can certainly discuss more in the Api View.

Reply:
Yes we can definitely discuss in the API view

Reply:
/cc @tg-msft as we discussed a similar consideration in the design of DelayStrategy - whether to make the method that users did not need to implement virtual or not.

Reply:
If I recall, I think we also talked about splitting the logic up that way in DelayStrategy so the non-virtual part forced standard behavior and the virtual part was all you needed to customize for your new strategy?  I don't feel strongly though and agree it's fine to hold off until the next Arch Board review for more opinions.

Reply:
Yes, that's right - but I think we have a similar situation here. If users implement TryValidate, they get the standard error handling via the Validate method. But as @jsquire pointed out, Validate needs to be virtual for mocking purposes.